### PR TITLE
Refactors Syndicate Bombs and Adds Chemical Bombs

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Refactors syndicate bombs, from TG and ports Chemical bombs

- Syndicate bombs tick more often
- Syndicate bombs can be fully deconstructed if they're disarmed and have no bomb core in them
- Observers can see a big timer hovering over the bomb (much like clonepods)
- Adds in cryo, pyro, and time released grenades
- Adds in syndicate chemical bombs

Pyro and cryo grenades are made in R&D, and, as their name implies, they release/suck in heat for the chemicals in their beakers.

Time release grenades are also made in R&D and release a little bit of their reagents at a time until the reagents are depleted.

Syndicate chemical bombs are a convoluted bombs that release chemicals over a wide area; they can be customized based on the the parts used to make them; they can hold 2-7 beakers (depending on the matter bin rating and the type of grenades used to make the payload) and its effects change based on the chemical grenades used to make it.

You can craft a core by combining C4 (or gibtonite) with a matter bin and two chemical grenades. You can make the frame by utilizing plasteel.

Code wise

Just a cleanup to make syndicate bombs use the fast processing system as opposed to regular processing; additionally, it cleans up some of the code and procs associated with syndicate bombs--such checking for all wires to be cut (as opposed to each one, individually with individual procs).

This also makes it easier to alter, in-game, how strong a syndicate bomb explosion is, as opposed to needing a "badmin" specific version to do so (just var edit the bomb, then var edit the bomb core).

:cl: Fox McCloud
add: Syndicate bombs now have visible timers for observers
add: Syndicate bombs tick more often
add: Syndnicate bombs can be deconstructed into plasteel if it is fully defused and has no bomb core
add: Can make Cryo, Pyro, and time released chemical grenades at R&D
add: Can make syndicate chemical bombs utilizing crafting
/:cl:

### CODE REVIEW

File: code/game/machinery/syndicatebomb.dm [hunk 1]
```diff
@@ -196,17 +287,26 @@
...
 
 /obj/item/weapon/bombcore/proc/detonate()
 	if(adminlog)
 		message_admins(adminlog)
 		log_game(adminlog)
-	explosion(get_turf(src),3,9,17, flame_range = 17)
-	if(src.loc && istype(src.loc,/obj/machinery/syndicatebomb/))
-		qdel(src.loc)
+	explosion(get_turf(src), range_heavy, range_medium, range_light, flame_range = range_flame)
+	if(loc && istype(loc,/obj/machinery/syndicatebomb/))
```
Comment (SamHPurp) [inline code/game/machinery/syndicatebomb.dm]: Does this need the trailing \/?

File: code/game/machinery/syndicatebomb.dm [hunk 2]
```diff
@@ -285,42 +388,43 @@
...
+	range_medium = 2
+	range_light = 4
+	range_flame = 2
 
 ///Syndicate Detonator (aka the big red button)///
 
 /obj/item/device/syndicatedetonator
 	name = "big red button"
-	desc = "Nothing good can come of pressing a button this garish..."
+	desc = "Your standard issue bomb synchronizing button. Five second safety delay to prevent 'accidents'"
```
Comment (SamHPurp) [inline code/game/machinery/syndicatebomb.dm]: Needs trailing `.`

**Comment by Fox-McCloud** [comment]
Screw it, implementing chemical bombs

File: code/game/machinery/syndicatebomb.dm [hunk 3]
```diff
@@ -406,6 +406,128 @@
...
+	desc = "An explosive payload designed to spread chemicals, dangerous or otherwise, across a large area. It is able to hold up to four chemical containers, and must be loaded before use."
+	origin_tech = "combat=4;materials=3"
+	icon_state = "chemcore"
+	var/list/beakers = list()
+	var/max_beakers = 1
+	var/spread_range = 5
+	var/temp_boost = 50
+	var/time_release = 0
+
+/obj/item/weapon/bombcore/chemical/detonate()
```
Comment (SamHPurp) [inline code/game/machinery/syndicatebomb.dm]: are there going to be any GC issues with the beakers on detonation?

**Comment by Rumiluntti** [comment]
How large is the area exactly if I decide to be an complete and utter dickwad and make 7 bluespess beakers of gibsmoke-foam hellmix in a chembomb?
Normal nade can delete a rather large area already, does this scale in some ridiculous manner or what?
Also, how fast does it spread the chems inside? Is it like an instant fuckhuge area of death?

**Pull request merged**